kornia/enhance/adjust.py

```diff
@@ -36,20 +37,21 @@ def adjust_saturation_raw(input: torch.Tensor, saturation_factor: Union[float, t
     See :class:`~kornia.color.AdjustSaturation` for details.
     """
 
-    if not torch.is_tensor(input):
+    if not isinstance(input, torch.Tensor):
         raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
 
     if not isinstance(saturation_factor, (float, torch.Tensor,)):
         raise TypeError(f"The saturation_factor should be a float number or torch.Tensor."
                         f"Got {type(saturation_factor)}")
 
     if isinstance(saturation_factor, float):
-        saturation_factor = torch.tensor([saturation_factor])
+        saturation_factor = torch.as_tensor(saturation_factor)
```

Comment:
this happens in line 50. It also forces that the saturation factor goes to same device and dtype as the input in case it comes as tensor. Maybe we can simplify this cases as you suggest and in case is a tensor with a different device or dtype as input, let pytorch raise an error. Might be easier later to detect potential bugs.